File: php/src/BillingService.php
<?php


class BillingService
{
    private $oldBillingSystem;

    /**
     * @param $oldBillingSystem
     */
    public function __construct(OldBillingSystem $oldBillingSystem)
    {
        $this->oldBillingSystem = $oldBillingSystem;
    }

    public function invoice(Invoice $invoice)
    {

        // Imagine loads of code here
        // ...
        $invoiceForBillingSystem = sprintf("%d%s to %s", $invoice->amount, $invoice->currency->symbol, $invoice->customer);
        $this->oldBillingSystem->invoice($invoiceForBillingSystem);
        // ...
        // Imagine loads of code here


    }
}

File: java/src/main/java/org/training/billing/BillingService.java
package org.training.billing;

public class BillingService {
    private final GoodOldBillingSystemAdapter goodOldBillingSystemAdapter;

    public BillingService(GoodOldBillingSystemAdapter goodOldBillingSystemAdapter) {
        this.goodOldBillingSystemAdapter = goodOldBillingSystemAdapter;
    }

    public void invoice(Invoice invoice) {
        int amount = invoice.amount;
        Currency currency = invoice.currency;
        String companyName = invoice.companyName;
        goodOldBillingSystemAdapter.invoice(amount, currency, companyName);
    }

}
